src/algorithms/paramspacesgd/paramspacesgd.jl

```diff
@@ -76,7 +15,17 @@ function init(rng::Random.AbstractRNG, alg::ParamSpaceSGD, q_init, prob)
     obj_st = init(rng, objective, adtype, q_init, prob, params, re)
     avg_st = init(averager, params)
     grad_buf = DiffResults.DiffResult(zero(eltype(params)), similar(params))
-    return ParamSpaceSGDState(prob, q_init, 0, grad_buf, opt_st, obj_st, avg_st)
+    if alg isa KLMinRepGradDescent
+        return KLMinRepGradDescentState(prob, q_init, 0, grad_buf, opt_st, obj_st, avg_st)
+    elseif alg isa KLMinRepGradProxDescent
+        return KLMinRepGradProxDescentState(
+            prob, q_init, 0, grad_buf, opt_st, obj_st, avg_st
+        )
+    elseif alg isa KLMinScoreGradDescent
+        return KLMinScoreGradDescentState(prob, q_init, 0, grad_buf, opt_st, obj_st, avg_st)
+    else
+        nothing
```

Comment:
Maybe throw a warning or error message here instead of letting it fail silently? 

```suggestion
        nothing
```